File: test/node/claude.js
#!/usr/bin/env node
"use strict";
// Test POST /v1/claude/chat
//
// Usage: node claude.js ["prompt"] [interactive_mode] [timeout]
//   prompt            default: "What is 2+2? Answer in one short sentence."
//   interactive_mode  accept | reject (default: reject)
//   timeout           seconds (default: 30)

const { postChat, printResult } = require("./_common");

async function main() {
  const prompt = process.argv[2] || "What is 2+2? Answer in one short sentence.";
  const interactiveMode = process.argv[3] || "reject";
  const timeout = Number(process.argv[4] || 30);

  const result = await postChat("/v1/claude/chat", {
    prompt,
    interactive_mode: interactiveMode,
    timeout,
  });
  printResult(result);
  process.exit(result.json.success ? 0 : 1);
}

main().catch((err) => {
  console.error("Request failed:", err.message);
  process.exit(1);
});

File: test/python/claude.py
#!/usr/bin/env python3
"""Test POST /v1/claude/chat

Usage: python3 claude.py ["prompt"] [interactive_mode] [timeout]
  prompt            default: "What is 2+2? Answer in one short sentence."
  interactive_mode  accept | reject (default: reject)
  timeout           seconds (default: 30)
"""
import sys

from _common import post_chat, print_result


def main() -> int:
    args = sys.argv[1:] + [""] * 3
    prompt = args[0] or "What is 2+2? Answer in one short sentence."
    interactive_mode = args[1] or "reject"
    timeout = int(args[2]) if args[2] else 30

    status, payload = post_chat(
        "/v1/claude/chat",
        {"prompt": prompt, "interactive_mode": interactive_mode, "timeout": timeout},
    )
    print_result(status, payload)
    return 0 if payload.get("success") else 1


if __name__ == "__main__":
    sys.exit(main())
